server/controllers/locations.js

```diff
@@ -86,7 +86,7 @@ var createFoursquareReq = function(type,params) {
 
 module.exports = {
   getLocations: function(req,res){
-
+    req.query.q = req.query.q.replace(/ /g, '_');
```

Comment:
q isn't a required param, so before replacing, there should be a check to see if q exists